brightray/common/main_delegate.cc

```diff
@@ -42,17 +42,17 @@ bool SubprocessNeedsResourceBundle(const std::string& process_type) {
 
 }  // namespace
 
-void InitializeResourceBundle(const std::string& locale) {
-  // Load locales.
-  ui::ResourceBundle::InitSharedInstanceWithLocale(
-      locale, nullptr, ui::ResourceBundle::DO_NOT_LOAD_COMMON_RESOURCES);
-
-  // Load other resource files.
+void LoadResourceBundle(const std::string& locale) {
+  if (!ui::ResourceBundle::HasSharedInstance())
+    ui::ResourceBundle::InitSharedInstanceWithLocale(
+        locale, nullptr, ui::ResourceBundle::DO_NOT_LOAD_COMMON_RESOURCES);
+  ui::ResourceBundle& bundle = ui::ResourceBundle::GetSharedInstance();
+  bundle.ReloadLocaleResources(locale);
```

Comment:
I think its better to cleanup the shared instance and reinitialize with the new locale if there is already a live instance rather than just `ReloadLocaleResources`.

```
const bool initialized = ui::ResourceBundle::HasSharedInstance();
if (initialized)
  ui::ResourceBundle::CleanupSharedInstance();
ui::ResourceBundle::InitSharedInstanceWithLocale(
    locale, nullptr, ui::ResourceBundle::DO_NOT_LOAD_COMMON_RESOURCES);
```

Reply:
What is the trade-off here -- what is the benefit in creating a new shared interest?

Not agreeing or disagreeing, just trying to understand the choice

Reply:
`InitSharedInstanceWithLocale` seems like a single point of initialization for resources, if chromium adds additional logic in the future we wouldn't have to worry about missing them while `ReloadLocaleResources` only reloads strings.

```
  // Changes the locale for an already-initialized ResourceBundle, returning the
  // name of the newly-loaded locale.  Future calls to get strings will return
  // the strings for this new locale.  This has no effect on existing or future
  // image resources.  |locale_resources_data_| is protected by a lock for the
  // duration of the swap, as GetLocalizedString() may be concurrently invoked
  // on another thread.
```